apps/assets/server/models/storage.js

```diff
@@ -19,7 +19,7 @@ const StorageSchema = new mongoose.Schema({
   metadata: {
     publicUrl: {
       type: String,
-      required: true
+      required: isPublicUrlRequired
```

Comment:
The public URL should always be required.
The scenario is when the given URL from the provider is different than the actual URL.

For example, assume I have my own domain, the I pass request from "images.mydomain.com" to my cloudinary account, using an "A RECORD" or "CNAME RECORD" to the cloudinary bucket, which can be something like "my-account-name.something.cloudinary.io".

From cloudinary's perspective, the base URL is their's, but from my own website, I want users to get images from my private domain (e.g. images.mydomain.com).

So this field is here to define my storage base URL, so when this service returns the metadata of an asset - its URL must be from the defined domain, and not cloudinary's.

Reply:
Oh ok. Got it. Thought this field was something else.

Reverted the StorageSchema publicUrl field to be required again